### PR TITLE
Convert dom to TypeScript

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

File: modules/angular2/src/facade/browser.ts [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 
-var win = window;
+export type ShadowRoot = DocumentFragment;
+export const ShadowRoot = (<any>global).ShadowRoot;
 
-export {win as window};
-export var document = window.document;
-export var location = window.location;
-export var gc = window.gc ? () => window.gc() : () => null;
+export {global as window};
```
Comment (vsavkin) [inline modules/angular2/src/facade/browser.ts]: This change is a bit weird.

It says at the top: `This library can only run in the browser.`, which means that you should be able to export `{win as window}`. It looks like we use browser.ts when running tests on node, which we probably should not.

Comment (mhevery) [inline modules/angular2/src/facade/browser.ts]: reverted

Comment (mhevery) [inline modules/angular2/src/facade/browser.ts]: OK, it broke the build. Looks like it is used in CJS, but it was like that before, I don't think I broke it.

**Comment by mhevery** [comment]
f302f70330b4327f1db2c19d34ceb48d62bdb4a8

**Pull request closed without merging**